src/iceberg/result.h

```diff
@@ -78,4 +78,12 @@ auto JsonParseError(const std::format_string<Args...> fmt, Args&&... args)
                             .message = std::format(fmt, std::forward<Args>(args)...)});
 }
 
+/// \brief Create an unexpected error with kInvalidExpression
+template <typename... Args>
+auto InvalidExpressionError(const std::format_string<Args...> fmt, Args&&... args)
+    -> unexpected<Error> {
+  return unexpected<Error>({.kind = ErrorKind::kInvalidExpression,
+                            .message = std::format(fmt, std::forward<Args>(args)...)});
+}
+
```

Comment:
Should we add an Error wrapper for each ErrorKind? I'd suggest including all those in this PR if needed. #80 might need NoSuchNamespaceError.

Reply:
That's a good idea! We can use a separate PR for this.